src/aero_model.F90

```diff
@@ -604,6 +604,88 @@ end subroutine aero_model_emissions
   ! private methods
   !=============================================================================
 
+  subroutine surf_area_dens( ncol, mmr, pmid, temp, beglev, endlev, sad, reff, sfc )
+    ! djlo : diam kept out as argument
+    use mo_constants     , only : pi
+!    use commondefinitions, only: nmodes_oslo => nmodes
+!    use const            , only: numberToSurface
+!    use aerosoldef       , only: lifeCycleNumberMedianRadius
+!    use aerosoldef       , only: lifeCycleSigma  ! djlo
+!    use oslo_utils       , only: calculateNumberConcentration
+
+    ! dummy args
+    integer,  intent(in)  :: ncol
+    real(r8), intent(in)  :: mmr(:,:,:)
+    real(r8), intent(in)  :: pmid(:,:)
+    real(r8), intent(in)  :: temp(:,:)
+    ! real(r8), intent(in)  :: diam(:,:,:) ! djlo : kept out as argument
+    integer,  intent(in)  :: beglev(:)
+    integer,  intent(in)  :: endlev(:)
+    real(r8), intent(out) :: sad(:,:)
+    real(r8), intent(out) :: reff(:,:)
+    real(r8),optional, intent(out) :: sfc(:,:,:)
+
+    ! local vars
+    !HAVE TO GET RID OF THIS MODE 0!! MESSES UP EVERYTHING!!
+    real(r8)         :: numberConcentration(pcols,pver,0:nmodes_oslo)
+    real(r8), target :: sad_mode(pcols,pver, nmodes_oslo)
+    real(r8)         :: vol_mode(pcols,pver, nmodes_oslo) ! djlo
+    real(r8)         :: vol(pcols,pver)                   ! djlo
+    real(r8) :: rho_air(pcols,pver)
+    integer :: m
+    integer :: i,k
+
+    ! Compute surface aero for each mode.
+    ! Total over all modes as the surface area for chemical reactions.
+
+    !Get air density in all layers
+    do k=1,pver
+       do i=1,ncol
+          rho_air(i,k) = pmid(i,k)/(temp(i,k)*287.04_r8)
+       end do
+    end do
+    !    
+    !Get number concentrations in all layers
+    call calculateNumberConcentration(ncol, mmr, rho_air, numberConcentration, isChemistry=.true.)
+
+    !Convert to area using lifecycle-radius
+    sad_mode = 0._r8
+    sad      = 0._r8
+    vol_mode = 0._r8
+    vol      = 0._r8
+    reff     = 0._r8
+
+    do i=1,ncol ! djlo : added explicit loop over i (because of ltrop(i) dependence
+       do k=beglev(i),endlev(i)
+          do m=1,nmodes_oslo
+             if ( m .eq. 1 &
+             .or. m .eq. 2 &
+             .or. m .eq. 4 &
+             .or. m .eq. 5 ) then
+! skipped mode 12 and 14
+!             .or. m .eq. 12 & 
+!             .or. m .eq. 14 ) then  
```

Comment:
I have removed the lines, but left a general comment in the code : 
currently modes 1, 2, 4, and 5 are used, but that it could be extended by modes 12 and 14.